sdk/textanalytics/ai-text-analytics/src/textAnalyticsResult.ts

```diff
@@ -128,3 +166,58 @@ export function makeTextAnalyticsErrorResult(
     error: intoTextAnalyticsError(error)
   };
 }
+
+/**
+ * combines successful and erroneous results into a single array of results and
+ * sort them so that the IDs order match that of the input documents array.
+ * @param input the array of documents sent to the service for processing.
+ * @param response the response received from the service.
+ */
+export function combineSuccessfulErroneousDocuments<T1 extends TextAnalyticsSuccessResult>(
```

Comment:
I dislike the name of this method, because instead of combining both successes and failures, it sounds like you're combining successful failures.

Reply:
renamed to `combineSuccessfulAndErroneousDocuments`.